src/google/adk/code_executors/gke_code_executor.py

```diff
@@ -0,0 +1,219 @@
+    """
+    namespace: str = "default"
+    image: str = "python:3.11-slim"
+    timeout_seconds: int = 300
+    cpu_limit: str = "500m"
+    mem_limit: str = "512Mi"
+
+    _batch_v1: client.BatchV1Api
+    _core_v1: client.CoreV1Api
+
+    def __init__(self, **data):
+        """Initializes the executor and the Kubernetes API clients.
+
+        This constructor supports overriding default class attributes (like
+        'namespace', 'image', etc.) by passing them as keyword arguments. It
+        automatically configures the Kubernetes client to work either within a
+        cluster (in-cluster config) or locally using a kubeconfig file.
+        """
+        super().__init__(**data)
+        try:
+            config.load_incluster_config()
+            logger.info("Using in-cluster Kubernetes configuration.")
+        except config.ConfigException:
+            logger.info("In-cluster config not found. Falling back to kubeconfig.")
+            config.load_kube_config()
+
+        self._batch_v1 = client.BatchV1Api()
+        self._core_v1 = client.CoreV1Api()
+
+    def execute_code(
+        self,
+        invocation_context: InvocationContext,
+        code_execution_input: CodeExecutionInput,
+    ) -> CodeExecutionResult:
+        """Orchestrates the secure execution of a code snippet on GKE."""
+        job_name = f"adk-exec-{uuid.uuid4().hex[:10]}"
+        configmap_name = f"code-src-{job_name}"
+
+        try:
+            self._create_code_configmap(configmap_name, code_execution_input.code)
+            job_manifest = self._create_job_manifest(job_name, configmap_name)
+
+            self._batch_v1.create_namespaced_job(
+                body=job_manifest, namespace=self.namespace
+            )
+            logger.info(f"Submitted Job '{job_name}' to namespace '{self.namespace}'.")
+            return self._watch_for_job_completion(job_name)
+
+        except ApiException as e:
+            logger.error(
+                "A Kubernetes API error occurred during job"
+                f" '{job_name}': {e.reason}",
+                exc_info=True,
+            )
+            return CodeExecutionResult(stderr=f"Kubernetes API error: {e.reason}")
+        except TimeoutError as e:
+            logger.error(e, exc_info=True)
+            logs = self._get_pod_logs(job_name)
+            stderr = f"Executor timed out: {e}\n\nPod Logs:\n{logs}"
+            return CodeExecutionResult(stderr=stderr)
+        except Exception as e:
+            logger.error(
+                f"An unexpected error occurred during job '{job_name}': {e}",
+                exc_info=True,
+            )
+            return CodeExecutionResult(stderr=f"An unexpected executor error occurred: {e}")
+        finally:
+            # The Job is cleaned up by the TTL controller, and we ensure the ConfigMap is always deleted.
+            self._cleanup_configmap(configmap_name)
+
+    def _create_job_manifest(self, job_name: str, configmap_name: str) -> client.V1Job:
```

Comment:
We should annotate the k8s resources, including the Job, with the invocation id from the `InvocationContext` to have some piece of data to link sandboxes to specific executions. 

There is an execution id (aka execution_session_id) in the execution context, but its only set for "stateful" executors, and its not clear what makes an executor stateful. The `stateful` is not really referenced anywhere from what I can tell.